packages/sdk/src/dataProtector/sharing/subgraph/getCollectionById.ts

```diff
@@ -0,0 +1,38 @@
+import { gql, type GraphQLClient } from 'graphql-request';
+import type { Address } from '../../types/index.js';
+
+export async function getCollectionById({
+  graphQLClient,
+  collectionTokenId,
+}: {
+  graphQLClient: GraphQLClient;
+  collectionTokenId: Address;
+}) {
+  const getProtectedDataQuery = gql`
+    query {
+      collection(id: "${collectionTokenId}") {
+        id
+        owner {
+          id
+        }
+        protectedDatas {
+          id
+        }
+        subscriptionParams {
+          duration
+          price
+        }
+      }
+    }
+  `;
+  const { collection } = await graphQLClient.request<{
+    collection: {
+      id: Address;
+      owner: { id: Address };
+      protectedDatas: { id: Address }[];
+      subscriptionParams: { duration: number; price: number };
+      subscriptions: { endDate: number }[];
+    };
+  }>(getProtectedDataQuery);
+  return collection;
```

Comment:
```suggestion
  if (!collection) {
    return null;
  }

  return {
    id: Number(collection.id),
    ...collection,
  };
```

Because we are more used to manipulate Numbers in JS, not bigints, hence converting it here.

Reply:
yeah well seen